### PR TITLE
Write delta gains

### PR DESCRIPTION
**Pull request by aewallwi**
Adds options to write out abscal degeneracies and also a basic multiplication script for dividing out and multiplying gains to extract the abscal degeneracy alone and also to reconstruct the full abscal solution from dgeneracies and recal files. Depends on #771 #772

### CODE REVIEW

File: hera_cal/abscal.py [hunk 1]
```diff
@@ -3526,6 +3527,8 @@ def post_redcal_abscal_run(data_file, redcal_file, model_files, raw_auto_file=No
         refant: tuple of the form (0, 'Jnn') indicating the antenna defined to have 0 phase. If None, refant will be automatically chosen.
         clobber: if True, overwrites existing abscal calfits file at the output path
         add_to_history: string to add to history of output abscal file
+        write_delta_gains: write the degenerate gain component solved by abscal so a separate file specified by output_file_delta
```
Comment (jsdillon) [inline hera_cal/abscal.py:3530]: I'd specific that the output file has flags and quality arrays that match the output abscal file

**Review by jsdillon — CHANGES_REQUESTED** [review]
Just a handful of minor tweaks and clarifications, but otherwise good to go.

File: hera_cal/abscal.py [hunk 2]
```diff
@@ -3690,9 +3693,61 @@ def post_redcal_abscal_run(data_file, redcal_file, model_files, raw_auto_file=No
...
+    gain_file_1: str
+        path to first gain file to multiply
+    gain_file_2: str
+        path to second gain file to multiply
+    output_file: str
+        path to output file.
+    clobber: bool, optional
+        overwrite existing output_file
+        default is False.
+    divide_gains: bool, optional
```
[the remark this answers is not in the document]
Comment (aewallwi) [reply hera_cal/abscal.py:3723]: This kwarg doesn't really have anything to do with the gain convention. It's just specifying whether we want to multiply the gains we input (for example, if the two gains are the abscal degen and a redcal solution) or if we want to divide the gains we input (for example, we want to extract the abscal degen and we input full abscal solution and redcal). I've added a comment to make sure this is clear.

File: hera_cal/abscal.py [hunk 3]
```diff
@@ -3690,9 +3693,61 @@ def post_redcal_abscal_run(data_file, redcal_file, model_files, raw_auto_file=No
...
+        divide gain 1 by gain 2
+        default is False.
+    """
+    hc1 = io.HERACal(gain_file_1)
+    hc1.read()
+    hc2 = io.HERACal(gain_file_2)
+    hc2.read()
+    if divide_gains:
+        hc1.gain_array = hc1.gain_array / hc2.gain_array
```
[the remark this answers is not in the document]
Comment (aewallwi) [reply hera_cal/abscal.py:3732]: No reason.

File: hera_cal/abscal.py [hunk 4]
```diff
@@ -3690,9 +3693,61 @@ def post_redcal_abscal_run(data_file, redcal_file, model_files, raw_auto_file=No
...
+    else:
+        hc1.gain_array *= hc2.gain_array
+    hc1.flag_array = hc1.flag_array | hc2.flag_array
+    hc1.total_quality_array = None
+    hc1.quality_array[:] = 0.
```
[the remark this answers is not in the document]
Comment (aewallwi) [reply hera_cal/abscal.py:3737]: Turns out `quality_array` is a required parameter. I've set this to `NaN` instead so as not to confuse anyone.

File: hera_cal/abscal.py (hunk 1 above)
Comment (aewallwi) [reply hera_cal/abscal.py:3530]: I've added this.

File: hera_cal/abscal.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (aewallwi) [reply hera_cal/abscal.py:3737]: See this line in uvcal https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/2ab5428b88739cba2b7e663b0346dab83a0f8ab6/pyuvdata/uvcal/uvcal.py#L711

**Review by jsdillon — APPROVED** [review]
ok, I'm good to go on this.

**Pull request merged**